nab/detectors/numenta/numentaNew_detector.py

```diff
@@ -0,0 +1,108 @@
+# ----------------------------------------------------------------------
+# Copyright (C) 2014, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+import math
+import simplejson as json
+
+from nupic.algorithms import anomaly_likelihood
+from nupic.frameworks.opf.common_models.cluster_params import (
+  getScalarMetricWithTimeOfDayAnomalyParams)
+from nupic.frameworks.opf.modelfactory import ModelFactory
+
+from nab.detectors.base import AnomalyDetector
+
+
+
+class NumentaNewDetector(AnomalyDetector):
+
+  def __init__(self, *args, **kwargs):
+
+    super(NumentaNewDetector, self).__init__(*args, **kwargs)
+
+    self.model = None
+    self.sensorParams = None
+    self.anomalyLikelihood = None
+
+
+  def getAdditionalHeaders(self):
+    """Returns a list of strings."""
+    return ["raw_score"]
+
+
+  def handleRecord(self, inputData):
+    """Returns a tuple (anomalyScore, rawScore).
+
+    Internally to NuPIC "anomalyScore" corresponds to "likelihood_score"
+    and "rawScore" corresponds to "anomaly_score". Sorry about that.
+    """
+    # Send it to NumentaNew detector and get back the results
+    result = self.model.run(inputData)
+
+    # Retrieve the anomaly score and write it to a file
+    rawScore = result.inferences["anomalyScore"]
+
+    # Compute log(anomaly likelihood)
+    anomalyScore = self.anomalyLikelihood.anomalyProbability(
+      inputData["value"], rawScore, inputData["timestamp"])
+    logScore = self.anomalyLikelihood.computeLogLikelihood(anomalyScore)
+
+    return (logScore, rawScore)
+
+
+  def initialize(self):
+    # Get config params, setting the RDSE resolution
+    rangePadding = abs(self.inputMax - self.inputMin) * 0.2
+    modelParams = getScalarMetricWithTimeOfDayAnomalyParams(
+      metricData=[0],
+      detectorType="numentaNew",
```

Comment:
"detector" is terminology specific to NAB. See [my comment in the nupic pr](https://github.com/numenta/nupic/pull/3138/files#r63962313).